python/sglang/srt/layers/moe/cutlass_moe.py

```diff
@@ -209,7 +209,7 @@ def cutlass_fused_experts_fp8(
     )
 
     result = torch.empty((m, k), device=device, dtype=out_dtype)
-    apply_shuffle_mul_sum(c2, result, c_map, topk_weights)
+    apply_shuffle_mul_sum(c2, result, c_map, topk_weights.to(out_dtype))
```

Comment:
Can we also add a dtype check in c++ side `void get_apply_shuffle_mul_sum_caller()` to avoid any misusage in the future?

Reply:
It seems we want to make sure the `c2`, `result`, `topk_weights` to have the same dtype, right?